probe-rs/src/architecture/arm/swo/mod.rs

```diff
@@ -12,19 +12,74 @@ pub enum SwoMode {
     Manchester,
 }
 
+#[derive(Debug)]
 pub struct SwoConfig {
     /// SWO mode: either UART or Manchester.
-    pub mode: SwoMode,
+    mode: SwoMode,
 
     /// Baud rate of SWO, in Hz.
     ///
     /// This value is used to configure what baud rate the target
     /// generates and to configure what baud rate the probe receives,
     /// so must be a baud rate supported by both target and probe.
-    pub baud: u32,
+    baud: u32,
 
     /// Clock input to TPIU in Hz. This is often the system clock (HCLK/SYSCLK etc).
-    pub tpiu_clk: u32,
+    tpiu_clk: u32,
+
+    /// Whether to enable TPIU formatting.
+    /// This is required to use ETM over SWO, but otherwise
+    /// adds overhead if only DWT/ITM data is used.
+    tpiu_continuous_formatting: bool,
+}
+
+impl SwoConfig {
+    /// Create a new SwoConfig using the specified TPIU clock in Hz.
+    ///
+    /// By default the UART mode is used at 1MBd and
+    /// TPIU continuous formatting is disabled (DWT/ITM only).
+    pub fn new(tpiu_clk: u32) -> Self {
+        SwoConfig {
+            mode: SwoMode::UART,
+            baud: 1_000_000,
+            tpiu_clk: tpiu_clk,
+            tpiu_continuous_formatting: false,
+        }
+    }
+
+    /// Set the baud rate in Hz.
+    pub fn set_baud(&mut self, baud: u32) -> &Self {
```

Comment:
Should we return 'Self' here? Right now you cannot chain config operations which would be great imo :) Or at least '&mut Self'?

Reply:
Yea, that would be better. At first I was going to have `enable_swo` consume the `SwoConfig` to make it clear it can't be updated later, but because both the probe `enable_swo` and the ARM `enable_swv` need the config (and I didn't want to make it Copy since that would defeat the point of passing it in), it seemed easier to keep them taking it by reference. I'll update so the setters return &mut though.